src/Form/inputTypeConfig.js

```diff
@@ -126,19 +126,21 @@ const inputTypeConfig = {
         },
         getIconAlternate: () => '',
         shouldDecorate: async (input, {isLogin, isHybrid, device}) => {
-            // if we are on a 'login' page, check if we have data to autofill the field
-            if (isLogin || isHybrid) {
-                return canBeAutofilled(input, device)
-            }
+            const subtype = getInputSubtype(input)
+            const variant = getInputVariant(input)
 
-            // at this point, it's not a 'login' form, so we could offer to provide a password
+            // Check first for password generation and the password.new scoring
             if (device.settings.featureToggles.password_generation) {
-                const subtype = getInputSubtype(input)
-                if (subtype === 'password') {
+                if (subtype === 'password' && variant === 'new') {
```

Comment:
I've moved this check first, because if we've scored it with `password.new` we don't care anymore the form type (those checks have already run), here we only care if feature toggle is set.